workflow-core/src/main/java/com/squareup/workflow1/StatefulWorkflow.kt

```diff
@@ -85,7 +85,13 @@ abstract class StatefulWorkflow<
     out RenderingT
     > : Workflow<PropsT, OutputT, RenderingT> {
 
-  inner class RenderContext internal constructor(
+  open inner class RenderContext internal constructor(
+    baseContext: BaseRenderContext<PropsT, StateT, OutputT>
+  ) : BaseRenderContext<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT> by baseContext
+
+  inner class RenderContextDeluxe internal constructor(
+    val props: @UnsafeVariance PropsT,
+    val state: StateT,
     baseContext: BaseRenderContext<PropsT, StateT, OutputT>
   ) : BaseRenderContext<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT> by baseContext
```

Comment:
Not sure why `RenderContext` needs to be open here, `RenderContextDeluxe` isn't subclassing it (and that's good – the less inheritance the better).

Reply:
I was just hacking and slashing to get something to work quickly.